source-map.bs

```diff
@@ -674,22 +676,84 @@ The index map follows the form of the standard map.  Like the regular source map
+    [=optionally report an error=].
+1. Let |sourceMap| be a new [=decoded source map=].
+1. Set |sourceMap|'s [=decoded source map/file=] to [=optionally get a string=] `"file"` from |jsonMap|.
+1. Let |previousOffset| be null.
+1. Let |previousLastMapping| be null.
+1. [=For each=] |section| of |jsonMap|[`"sections"`]:
+    1. If |section| is not a [=/map=], [=optionally report an error=] and continue with the next iteration.
+    1. If |section|[`"offset"`] does not [=map/exist=], then report an error.
+    1. Let |offsetLine| be |section|[`"offset"`][`"line"`].
+    1. Let |offsetColumn| be |section|[`"offset"`][`"column"`].
+    1. If |offsetLine| is not a number, [=optionally report an error=].
+    1. If |offsetColumn| is not a number, [=optionally report an error=].
+    1. If |previousOffset| is not null,
+        1. If |previousLastMapping| is not null,
+            1. If |offsetLine| is less than |previousLastMapping|'s [=decoded mapping/generatedLine=], [=optionally report an error=].
+            1. If |offsetLine| is equal to |previousLastMapping|'s [=decoded mapping/generatedLine=] and |offsetColumn| is less than |previousLastMapping|'s [=decoded mapping/generatedColumn=], [=optionally report an error=].
+        1. Otherwise,
+            1. If |offsetLine| is less than |previousOffset|[`"line"`], [=optionally report an error=].
+            1. If |offsetLine| is equal to |previousOffset|[`"line"`] and |offsetColumn| is less than |previousOffset|[`"column"`], [=optionally report an error=].
+
+        Note: This part of the decoding algorithm checks that index source map sections do not overlap. While it is
+              expected that generators should not produce index source maps with overlapping sections, source map
+              consumers may choose instead to only check the simpler condition that the sections are ordered and therefore
+              that offsets appear in increasing order.
+
+    1. If |section|[`"map"`] does not [=map/exist=], then report an error.
+    1. Let |decodedSection| be the result of [=decoding a source map=] given |section|[`"map"`] and |baseURL|.
+    1. If the previous step reported an error, [=optionally report an error=] and continue with the next iteration.
+    1. Set |sourceMap|'s [=decoded source map/sources=] to the result of [=list/extending=] |sourceMap|'s [=decoded source map/sources=] with |decodedSection|'s [=decoded source map/sources=].
+    1. Let |offsetMappings| be a new [=list=].
+    1. [=For each=] |mapping| of |decodedSection|'s [=decoded source map/mappings=]:
+        1. Let |offsetMapping| be a new [=decoded mapping=].
+        1. Set |offsetMapping|'s [=decoded mapping/generatedLine=] to |mapping|'s [=decoded mapping/generatedLine=] + |offsetLine|.
+        1. Set |offsetMapping|'s [=decoded mapping/generatedColumn=] to |mapping|'s [=decoded mapping/generatedColumn=] + |offsetColumn|.
```

Comment:
This looks like it also adds the `offsetColumn` to all subsequent lines? Should this be guarded by "if this is the first generated line of the mapping"?

Reply:
Oh thanks for catching this, yeah this should only apply to the first line.

(alternatively, this part of the algorithm could be changed to pass the offsets to the "decode a source map" as initial offset arguments which are normally 0)